---
title: "Controlled recombination simulation, model analysis, cycle 6 & 10"
output: github_document
always_allow_html: true
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(dpi=400,fig.width=7)
```

### Notes:   
Each simulation rep draws from 26 founders, 10 cycles of phenotypic selection burnin 400 F1s -> 400 DHs, phenotype, serves as TP and GS cycle 1 starting pop, variance standardized to 1 -> change recombination distribution (WT, 20X Peri or 20X Chr) --> 400 F1s -> 400 DHs, train model, phenotype and update TP with top 160 (40%) DHs and drop bottom 80 (20%), advance top 20 DHs (5% selection intensity) for next cycle of 400 new F1s, repeat for 10 cycles   

**WT map type does not get a recombination change, WT is always WT while Peri and Chr are always 2 or 20X**  

<details><summary>Click here: simulation parameters</summary> 

#### Variables:   

**Recombination:** WT, 2X or 20X     
**Genetic map change:** WT, Pericentromere or Chromosome   
**Repulsion:** 1:5  
**QTL type:** Random or deleterious variant 
**Heritability:** 0.2 or 0.8   
**QTL per Chr:** 2 or 200    
**Relationship matrix:** genomewide or causal variant  

#### Load packages and data
```{r load data, message=FALSE, warning=FALSE}
library(tidyverse)
library(data.table)
library(kableExtra)
library(ggsci)
library(gt)
library(lme4)
library(emmeans)
library(car)
library(broom.mixed)

gg <- fread("https://raw.githubusercontent.com/etaagen/dissertation_chapter_4/main/Supplementary_2/results_S2.1/f_gg.csv") %>% as.data.frame()
gv <- fread("https://raw.githubusercontent.com/etaagen/dissertation_chapter_4/main/Supplementary_2/results_S2.1/f_gv.csv") %>% as.data.frame()
pa <- fread("https://raw.githubusercontent.com/etaagen/dissertation_chapter_4/main/Supplementary_2/results_S2.1/f_pa.csv") %>% as.data.frame()
be <- fread("https://raw.githubusercontent.com/etaagen/dissertation_chapter_4/main/Supplementary_2/results_S2.1/f_be.csv") %>% as.data.frame()
# be `value` is 2-fold higher than it should be (ASR does not know how to treat VarA/GenicVarA for DH)
be$value <- be$value/2
qtl <- fread("https://raw.githubusercontent.com/etaagen/dissertation_chapter_4/main/Supplementary_2/results_S2.1/f_qtl.csv") %>% as.data.frame()
qtl_af <- fread("https://raw.githubusercontent.com/etaagen/dissertation_chapter_4/main/Supplementary_2/results_S2.1/f_qtl_af.csv") %>% as.data.frame()

```


**Data frames**    
 
gg: population's genetic gain   
gv: additive genetic variance of population   
pa: prediction accuracy of genomic selection  
be: Bulmer effect (varA/genicVarA)  
qtl: positive and negative effect QTL fixation ratio    
qtl_af: change in QTL negative allele frequency, subset by small (Q1), medium, or large (Q3) effect size 

**Columns**  
  
**rep** is 1:number of reps simulation was run    
**cycle** designates founder / burnin (0), and GS cycle 1:10   
**value** is the raw response variable measurement, (or average after summary function, grouped by legend and cycle, of all rep)    
**Matrix** is the relationship matrix used in RRBLUP, genomewide or causal variant   
**Pop** is the full founder set (F) or biparental (bp)  
**Recombination** is the scale of map change, 2 or 20X   
**H2** is the broad sense heritability   
**QTL** is the number of QTL per chromosome  
**QTL_type** is R or DV, random or deleterious variant  
**Map_Type** is the WT, Pericentromere, or Chromosome-wide change to the genetic map, given a Recombination scale  
**Repulsion** is 1:5 representing different coupling and replusion ratios  

* 1: Additive effect signs are positive for all QTL (select against minor allele)  

* 2: Random 2/3 of additive effect signs are positive and 1/3 are negative for QTL   

* 3: Random 1/2 of additive effect signs are positive and 1/2 are negative for QTL   

* 4: 1/2 of additive effect signs are positive and 1/2 are negative for QTL, alternating positive or negative each QTL   

* 5: Random 1/3 of additive effect signs are positive and 2/3 are negative for QTL (most selection for minor allele)   

</details> 

### Genetic gain  

**Linear model, ANOVA table** 

<details><summary>Click here: Model, ANOVA tables cycle 6 & 10</summary>

**Note:** filtered for cycle 6 or cycle 10 observations. See .Rmd file for code.  

`response variable ~ (map type + recombination + QTL per Chr + H2 + repulsion + matrix + QTL type)^2 + (1|rep)`   

 
```{r, echo=FALSE}
 # set factors
my_data <- gg
my_data$rep <- as.factor(my_data$rep)
my_data$Map_type <- as.factor(my_data$Map_type)
my_data$Recombination<- as.factor(my_data$Recombination)
my_data$QTL<- as.factor(my_data$QTL)
my_data$H2<- as.factor(my_data$H2)
my_data$Repulsion<- as.factor(my_data$Repulsion)
my_data$Matrix<- as.factor(my_data$Matrix)
my_data$QTL_type <- as.factor(my_data$QTL_type)

# filter to 6th and 10th cycle
my_data_6 <- my_data %>% filter(cycle == 6)
my_data_10 <- my_data %>% filter(cycle == 10)

# run mixed model 
my_lmer_6 <- lmer(value ~ (Map_type + Recombination+QTL+H2+Repulsion+Matrix+QTL_type)^2 + (1|rep), data = my_data_6)
my_lmer_10 <- lmer(value ~ (Map_type + Recombination+QTL+H2+Repulsion+Matrix+QTL_type)^2 + (1|rep), data = my_data_10)

```
  
  
ANOVA table, cycle 6:  
```{r, echo=FALSE, message=FALSE}
# run Anova
my_Anova_6 <- Anova(my_lmer_6) 
# print the 6th cycle of selection ANOVA table
print(my_Anova_6)
```
  
ANOVA table, cycle 10:  
```{r, echo=FALSE, message=FALSE}
# run Anova
my_Anova_10 <- Anova(my_lmer_10) 
# print the 10th cycle of selection ANOVA table
print(my_Anova_10)
```
  

 </details>




### Genetic variance

**Linear model,  ANOVA table**  

<details><summary>Click here: Model, ANOVA tables cycle 6 & 10</summary> 

**Note:** filtered for cycle 6 or cycle 10 observations. See .Rmd file for code.  

`response variable ~ (map type + recombination + QTL per Chr + H2 + repulsion + matrix + QTL type)^2 + (1|rep)`  


 
```{r, echo=FALSE}
 # set factors
my_data <- gv
my_data$rep <- as.factor(my_data$rep)
my_data$Map_type <- as.factor(my_data$Map_type)
my_data$Recombination<- as.factor(my_data$Recombination)
my_data$QTL<- as.factor(my_data$QTL)
my_data$H2<- as.factor(my_data$H2)
my_data$Repulsion<- as.factor(my_data$Repulsion)
my_data$Matrix<- as.factor(my_data$Matrix)
my_data$QTL_type <- as.factor(my_data$QTL_type)

# filter to 6th and 10th cycle
my_data_6 <- my_data %>% filter(cycle == 6)
my_data_10 <- my_data %>% filter(cycle == 10)

# run mixed model 
my_lmer_6 <- lmer(value ~ (Map_type + Recombination+QTL+H2+Repulsion+Matrix+QTL_type)^2 + (1|rep), data = my_data_6)
my_lmer_10 <- lmer(value ~ (Map_type + Recombination+QTL+H2+Repulsion+Matrix+QTL_type)^2 + (1|rep), data = my_data_10)

```
  
  
ANOVA table, cycle 6:  
```{r, echo=FALSE, message=FALSE}
# run Anova
my_Anova_6 <- Anova(my_lmer_6) 
# print the 6th cycle of selection ANOVA table
print(my_Anova_6)
```
  
ANOVA table, cycle 10:  
```{r, echo=FALSE, message=FALSE}
# run Anova
my_Anova_10 <- Anova(my_lmer_10) 
# print the 10th cycle of selection ANOVA table
print(my_Anova_10)
```
  
</details>   



  

### Bulmer effect  

**Linear model,  ANOVA table**  

<details><summary>Click here: Model, ANOVA tables cycle 6 & 10</summary>

**Note:** filtered for cycle 6 or cycle 10 observations. See .Rmd file for code.  

`response variable ~ (map type + recombination + QTL per Chr + H2 + repulsion + matrix + QTL type)^2 + (1|rep)`   

 
```{r, echo=FALSE}
 # set factors
my_data <- be
my_data$rep <- as.factor(my_data$rep)
my_data$Map_type <- as.factor(my_data$Map_type)
my_data$Recombination<- as.factor(my_data$Recombination)
my_data$QTL<- as.factor(my_data$QTL)
my_data$H2<- as.factor(my_data$H2)
my_data$Repulsion<- as.factor(my_data$Repulsion)
my_data$Matrix<- as.factor(my_data$Matrix)
my_data$QTL_type <- as.factor(my_data$QTL_type)

# filter to 6th and 10th cycle
my_data_6 <- my_data %>% filter(cycle == 6)
my_data_10 <- my_data %>% filter(cycle == 10)

# run mixed model 
my_lmer_6 <- lmer(value ~ (Map_type + Recombination+QTL+H2+Repulsion+Matrix+QTL_type)^2 + (1|rep), data = my_data_6)
my_lmer_10 <- lmer(value ~ (Map_type + Recombination+QTL+H2+Repulsion+Matrix+QTL_type)^2 + (1|rep), data = my_data_10)

```
  
  
ANOVA table, cycle 6:  
```{r, echo=FALSE, message=FALSE}
# run Anova
my_Anova_6 <- Anova(my_lmer_6) 
# print the 6th cycle of selection ANOVA table
print(my_Anova_6)
```
  
ANOVA table, cycle 10:  
```{r, echo=FALSE, message=FALSE}
# run Anova
my_Anova_10 <- Anova(my_lmer_10) 
# print the 10th cycle of selection ANOVA table
print(my_Anova_10)
```
    

</details>   




### Prediction accuracy

**Linear model,  ANOVA table**  

<details><summary>Click here: Model, ANOVA tables cycle 6 & 10</summary>

**Note:** filtered for cycle 6 or cycle 10 observations. See .Rmd file for code.   

`response variable ~ (map type + recombination + QTL per Chr + H2 + repulsion + matrix + QTL type)^2 + (1|rep)`  


 
```{r, echo=FALSE}
 # set factors
my_data <- pa
my_data$rep <- as.factor(my_data$rep)
my_data$Map_type <- as.factor(my_data$Map_type)
my_data$Recombination<- as.factor(my_data$Recombination)
my_data$QTL<- as.factor(my_data$QTL)
my_data$H2<- as.factor(my_data$H2)
my_data$Repulsion<- as.factor(my_data$Repulsion)
my_data$Matrix<- as.factor(my_data$Matrix)
my_data$QTL_type <- as.factor(my_data$QTL_type)

# filter to 6th and 10th cycle
my_data_6 <- my_data %>% filter(cycle == 6)
my_data_10 <- my_data %>% filter(cycle == 10)

# run mixed model 
my_lmer_6 <- lmer(value ~ (Map_type + Recombination+QTL+H2+Repulsion+Matrix+QTL_type)^2 + (1|rep), data = my_data_6)
my_lmer_10 <- lmer(value ~ (Map_type + Recombination+QTL+H2+Repulsion+Matrix+QTL_type)^2 + (1|rep), data = my_data_10)

```
  
  
ANOVA table, cycle 6:  
```{r, echo=FALSE, message=FALSE}
# run Anova
my_Anova_6 <- Anova(my_lmer_6) 
# print the 6th cycle of selection ANOVA table
print(my_Anova_6)
```
  
ANOVA table, cycle 10:  
```{r, echo=FALSE, message=FALSE}
# run Anova
my_Anova_10 <- Anova(my_lmer_10) 
# print the 10th cycle of selection ANOVA table
print(my_Anova_10)
```
    
 
</details>  




### QTL fixation   

**Linear model has `singular fit`, omitted**    

### QTL allele frequency change 

**Linear model,  ANOVA table**  

<details><summary>Click here: Model, ANOVA table</summary>

**Note:** total cycle change observations. See .Rmd file for code.  

`response variable ~ (map type + recombination + QTL per Chr + H2 + repulsion + matrix + QTL type + allele)^2 + (1|rep)` 


```{r, echo=FALSE}
 # set factors
my_data <- qtl_af
my_data$rep <- as.factor(my_data$rep)
my_data$Map_type <- as.factor(my_data$Map_type)
my_data$Recombination<- as.factor(my_data$Recombination)
my_data$QTL<- as.factor(my_data$QTL)
my_data$H2<- as.factor(my_data$H2)
my_data$Repulsion<- as.factor(my_data$Repulsion)
my_data$Matrix<- as.factor(my_data$Matrix)
my_data$QTL_type <- as.factor(my_data$QTL_type)

names(my_data)[1] <- "Allele"
my_data$Allele <- as.factor(my_data$Allele)

my_data <- my_data %>% filter(Allele %in% c("large_pos", "med_pos", "small_pos"))

# run mixed model 
my_lmer <- lmer(value ~ (Map_type + Recombination+QTL+H2+Repulsion+Matrix+QTL_type+Allele)^2 + (1|rep), data = my_data)

```
  
  
ANOVA table:  
```{r, echo=FALSE, message=FALSE}
# run Anova
my_Anova <- Anova(my_lmer) 
# print ANOVA table
print(my_Anova)
```
  
</details>


